R/package_check.R

```diff
@@ -0,0 +1,61 @@
+
+#' Package Check
+
+#' @description
+#' This function checks the status of a specified package in the current R environment. It verifies whether the package is installed, and if so, it compares the installed version with the latest version available online.
+
+#' @param package A character string specifying the name of the package to be checked.
+
+#' @return A numeric vector with the following elements:
+#' \itemize{
+#'   \item \code{[1]} Status code:
+#'     \itemize{
+#'       \item \code{0} - Package not found (incorrect spelling).
+#'       \item \code{1} - Package found and installed.
+#'       \item \code{2} - Package found, but not installed.
+#'     }
+#'   \item \code{[2]} Version comparison result:
+#'     \itemize{
+#'       \item \code{-1} - Installed version is older than the latest version.
+#'       \item \code{0} - Installed version is the same as the latest version.
+#'       \item \code{1} - Installed version is newer than the latest version.
+#'     }
+#'   \item \code{[3]} Installed version (if available), as a character string.
+#'   \item \code{[4]} Latest version available online (if available), as a character string.
+#' }
```

Comment:
I'm intrigued to see how this comes out. I'm not sure at all, personally, so I've just made a mental note to check this once it's merged. We might come back to this.